PowerUp/app/src/main/java/powerup/systers/com/ui/avatarRoom/AvatarRoomPresenter.java

```diff
@@ -0,0 +1,86 @@
+package powerup.systers.com.ui.avatarRoom;
+
+import android.content.Context;
+
+import powerup.systers.com.R;
+import powerup.systers.com.data.DataSource;
+
+public class AvatarRoomPresenter implements AvatarRoomContract.IAvatarRoomPresenter {
+    private AvatarRoomContract.IAvatarRoomView view;
+    private DataSource source;
+    private Context context;
+
+    public AvatarRoomPresenter(AvatarRoomContract.IAvatarRoomView view, DataSource source, Context context) {
+        this.view = view;
+        this.source = source;
+        this.context = context;
+    }
+
+    @Override
+    public void setValues(int eye, int hair, int skin, int cloth) {
+        calculateEyeValue(eye);
+        calculateHairValue(hair);
+        calculateSkinValue(skin);
+        calculateClothValue(cloth);
+    }
+
+    @Override
+    public void calculateEyeValue(int value) {
+        String eyeImageName = context.getResources().getString(R.string.eye);
+        eyeImageName = eyeImageName + value;
+        R.drawable ourRID = new R.drawable();
+        java.lang.reflect.Field photoNameField;
+        try {
+            photoNameField = ourRID.getClass().getField(eyeImageName);
+            view.updateAvatarEye(photoNameField.getInt(ourRID));
+        } catch (NoSuchFieldException | IllegalAccessException
+                | IllegalArgumentException e) {
+            e.printStackTrace();
```

Comment:
have more info, while logging exception here, like adding value, which caused the function to get Exception. (fast/easy debugging)

Reply:
just take care, this at all the places.